docs/api.md

```diff
@@ -0,0 +1,253 @@
+# soundslike - API documentation
+Learn how to use the soundslike API!
+
+First. for the sake of this, im going to use soundslike.com as the url, however localhost:8888 might be the url if you're using docker 
+
+## Section 1. Authentication 
+
+### Getting auth token from api
+
+run 'GET' http://soundslike.com/api/token with a username and password with basic AUTH  
+
+this will return a response of...
+```json
+    {
+        "expiration": 3000,
+        "token": "token_string"
+    }
+``` 
+you'll need to Base64 encode the bearer token provided earlier and include it in all API requests, except user creation 
+
+## Section 2. Users 
+
+### creating a new user 
+
+run 'POST' http://soundslike.com/api/users/
+
+with a application/json body of...
+```json
+{
+    "username" : "test", 
+    "email" : "test@gmail.com", 
+    "password": "test"
+}
+``` 
+and this will return a response of... 
+```json
+{
+    "email": "test@gmail.com",
+    "member_since": "Fri, 07 Dec 2018 06:19:28 GMT",
+    "username": "test"
+}
+``` 
+### getting a user 
+
```

Comment:
For all endpoints, I think we should include a generic version of the endpoint to indicate what type of field needs to be passed in. So in this case it would be `/api/users/<username: string>` or something like that. something that looks really similar to the route decorators basically.